ifrs17/Utils/ActivityLogsAndImports.ipynb

```diff
@@ -0,0 +1,302 @@
+                "\n        string format;",
+                "\n        using (MemoryStream ms = new MemoryStream())",
+                "\n        {",
+                "\n            await stream.Stream.CopyToAsync(ms);",
+                "\n            content = ms.ToArray();",
+                "\n            ms.Position = 0;",
+                "\n            DataSetImportVariable importVariable = new DataSetImportVariable(session, sess => sess.CancellationToken);",
+                "\n            var dsRes = await importVariable.ReadFromStream(ms).ExecuteAsync();",
+                "\n            format = dsRes.Format;",
+                "\n            if (stream.WillBeReused)",
+                "\n                stream.Stream.Position = 0;",
+                "\n            else",
+                "\n            {",
+                "\n                stream.Stream.Close();",
+                "\n                await stream.Stream.DisposeAsync();",
+                "\n            }",
+                "\n        }",
+                "\n        return (format, content);",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record KeyedFile : ImportWithKey",
+                "\n{",
+                "\n    public string Name {get; init;}",
+                "\n",
+                "\n    public string Directory {get; init;}",
+                "\n    ",
+                "\n    public string ContentType {get; init;}",
+                "\n",
+                "\n    [Conversion(typeof(JsonConverter<string[]>))]",
+                "\n    public string[] Partition {get; init;}",
+                "\n",
+                "\n    public string Source {get; init;}",
+                "\n",
+                "\n    public KeyedFile(FileImportOptions options)",
+                "\n    {",
+                "\n        importOptions = options;",
+                "\n        string fileName = options.FileName; ",
+                "\n        Id = Guid.NewGuid();",
+                "\n        Name = Path.GetFileName(fileName);",
+                "\n        Directory = Path.GetDirectoryName(fileName); ",
+                "\n        ContentType = Path.GetExtension(fileName);",
+                "\n        Source = options.Storage.GetType().Name; ",
+                "\n        Partition =  GetInvolvedPartitions(options);",
+                "\n        // Andrey Katz: Options.TargetDataSource.Partion.GetCurrent(?? What do we put here, different classes might posess various partitions, e.g. Yield Curve has none ??)",
+                "\n    }",
+                "\n",
+                "\n    private string[] GetInvolvedPartitions(ImportOptions options)",
+                "\n    {",
+                "\n        //Andrey Katz: Get all the relevant partitions here ",
```

Comment:
TODO, right?

Reply:
Yep, it is TODO. I marked it now as TODO explicitly